homeassistant/util/package.py

```diff
@@ -40,14 +41,32 @@ def is_installed(requirement_str: str) -> bool:
     expected input is a pip compatible package specifier (requirement string)
     e.g. "package==1.0.0" or "package>=1.0.0,<2.0.0"
 
+    For backward compatibility, it also accepts a URL with a fragment
+    e.g. "git+https://github.com/pypa/pip#pip>=1"
+
     Returns True when the requirement is met.
     Returns False when the package is not installed or doesn't meet req.
     """
     try:
         req = Requirement(requirement_str)
     except InvalidRequirement:
-        _LOGGER.error("Invalid requirement '%s'", requirement_str)
-        return False
+        if "#" not in requirement_str:
```

Comment:
I was hoping we could always call urlparse if there is a `#` but pep 0508 allows # in env markers so we cannot assume # means its a fragment and have to fallback